### PR TITLE
Reduce logging in tests by using slf4j-api instead of slf4j-simple, i…

### LINKED ISSUE #305: Reduce spark logging in tests
**Description**

Reduce spark logging in tests

**Use case or motivation behind the feature request**

It prints a lot of irrelevant logs which slows down usage and drowns possibly interesting messages.
I think the default logging level can be set to WARN+ for spark components long as it can be changed on demand

**Related issues**

**Additional context**

Any build log from actions

### PR DESCRIPTION
**Pull request by StrongestNumber9**
…ntroduce new log4j2-test.xml

Note: Currently prints warning about multiple slf4j providers, but only once.

```
[INFO] Running com.teragrep.pth10.ReplaceTransformationTest
SLF4J: Class path contains multiple SLF4J providers.
SLF4J: Found provider [org.apache.logging.slf4j.SLF4JServiceProvider@2dfaea86]
SLF4J: Found provider [org.slf4j.jul.JULServiceProvider@15888343]
SLF4J: See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual provider is of type [org.apache.logging.slf4j.SLF4JServiceProvider@2dfaea86]
```

Fixes #305

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -241,9 +241,8 @@
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-simple</artifactId>
-      <version>1.7.36</version>
-      <scope>test</scope>
+      <artifactId>slf4j-api</artifactId>
+      <version>2.0.16</version>
```
Comment (kortemik) [inline pom.xml:245]: has no scope, is it possible to use test scope?

File: src/test/resources/log4j2-test.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <Appenders>
+    <Console name="Console" target="SYSTEM_OUT">
+      <PatternLayout pattern="%5p [%d] [%C] ({%t} %F[%M]:%L) - %m%n"/>
+    </Console>
+  </Appenders>
+  <Loggers>
+    <Logger name="com.teragrep" level="info">
+      <AppenderRef ref="Console"/>
+    </Logger>
+    <Root level="warn">
```
Comment (kortemik) [inline src/test/resources/log4j2-test.xml:12]: is it possible to use default info but org.apache.spark.internal.Logging to warn?

Comment (StrongestNumber9) [reply src/test/resources/log4j2-test.xml]: Seems like so, updated

File: pom.xml (hunk 1 above)
Comment (StrongestNumber9) [reply pom.xml:245]: Not possible to use test scope unless org.slf4j Logger gets provided via some route

DedupTransformation.java:[60,17] package org.slf4j does not exist
DedupTransformation.java:[68,26] cannot find symbol
[ERROR]   symbol:   class Logger
[ERROR]   location: class com.teragrep.pth10.ast.commands.transformstatement.DedupTransformation

**Review by kortemik — APPROVED** [review]

**Pull request merged**